### PR TITLE
989 - IdsDataGrid: Allow custom selected cell colors

### LINKED ISSUE #989: IdsDataGrid: Show a custom background color and custom selection color
**Task**

M3 has the ability to customize row and cell color. We can do this via the cssPart feature as shown on example https://main.wc.design.infor.com/ids-data-grid/columns-custom-css.html

However, if they add selection the row selection color also needs to be customizable. So extend this example to show selection and add another setting(s) to set the custom color of the selected rows with another css part.

Should also have a callback s it can be set dynamically (takes a string or a function like the current one)

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds a new IdsDataGridColumn setting, `cellSelectedCssPart`, which optionally allows a column to provide a dynamic [CSS part]() name that can be used to provide a custom background color -- or other CSS -- to selected/activated grid cells.

**Related github/jira issue (required)**:
Closes #989

**Steps necessary to review your pull request (required)**:
- Pull/build/run
- Open http://localhost:4300/ids-data-grid/columns-custom-css.html
- Click a cell without selecting the row (using the checkbox).  When the row is activated, slightly darker custom colors should appear on the cell.
- Select some rows using the checkboxes.   When the rows are selected, the same darker background colors are applied.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: test/ids-data-grid/ids-data-grid-func-test.ts [hunk 1]
```diff
@@ -2478,25 +2483,4 @@ describe('IdsDataGrid Component', () => {
       expect(dataGrid.getAttribute('id-column')).toBeFalsy();
     });
   });
-
-  describe('Events Tests', () => {
```
Comment (andriivarhanov) [inline test/ids-data-grid/ids-data-grid-func-test.ts:2482]: what's the reason to remove the test?

Comment (EdwardCoyle) [reply test/ids-data-grid/ids-data-grid-func-test.ts:2482]: Good question... I think I accidentally got rid of these during my last rebase.  Good catch 👍

**Review by andriivarhanov — APPROVED** [review]

**Review by tmcconechy — CHANGES_REQUESTED** [review]
@EdwardCoyle just wondering about a few things :

1. The price column when activated looks slate? I guess this is just the example but can we make it like the other colors? makes it look not selected
2. Something seems wrong with the way it works for mixed selection. I.E. when i click the checkbox it should only activate the checkbox, clicking the row activates (its now activating and marking the checkbox) see http://localhost:4300/ids-data-grid/mixed-selection.html
3. Open src/components/ids-data-grid/demos/columns-custom-css.ts and you get a type error (or run `npx tsc`). I think just need to add the new setting cellSelectedCssPart to IdsDataGridColumn

**Comment by tmcconechy** [comment]
@EdwardCoyle 

1. I changed it - so it has the same color -> super minor but just for esthetics
2. Yes i see it is a little incorrect for mixed -> click just the checkbox and in this case it shouldn't activate the row like http://localhost:4300/ids-data-grid/mixed-selection.html. But that being said  i think it would be ok to use multiple select in the example, then its correct. 
3. I dont see this anymore, sorry must have gotten my branches mixed

**Comment by EdwardCoyle** [comment]
@tmcconechy I think I fixed the row highlighting logic when in mixed selection mode now... I misunderstood what you meant by "activate" since it seemed everything was working (just that highlighting the row was occurring on "select" in mixed selection mode when it shouldn't).  Can you take another look to confirm?

**Review by andriivarhanov — APPROVED** [review]

File: src/components/ids-data-grid/ids-data-grid.ts [hunk 2]
```diff
@@ -1634,6 +1642,38 @@ export default class IdsDataGrid extends Base {
...
+
+    const cells = row.querySelectorAll('.ids-data-grid-cell');
+    if (cells?.length) {
+      [...cells].forEach((cell: Element, columnIndex: number) => {
+        const columnData = this.columns[columnIndex];
+        let cssPart = columnData.cssPart || 'cell';
+
+        // Updates selected rows to display the correct CSS part (also activated rows in mixed-selection mode)
+        if (
+          (this.rowSelection === 'mixed' && row.classList.contains('activated'))
```
Comment (clayinfor) [inline src/components/ids-data-grid/ids-data-grid.ts:1662]: These conditionals could be made more readable as follows:
```javascript
const isMixed = this.rowSelection === 'mixed';
const isNotMixed = !isMixed; // this.rowSelection === 'single' || this.rowSelection === 'multiple'; 
const isActivatedAndMixed = row.classList.contains('activated') && isMixed;
const isSelectedAndNotMixed = row.classList.contains('selected') && isNotMixed;

if (isMixedAndActivated || isSelectedAndNotMixed) {
  cssPart = columnData.cellSelectedCssPart || 'cell-selected';
}
```

**Review by clayinfor — COMMENTED** [review]
👍

**Pull request merged**